generic_server.cc

```diff
@@ -321,7 +324,22 @@ server::listen(socket_address addr, std::shared_ptr<seastar::tls::credentials_bu
 future<> server::do_accepts(int which, bool keepalive, socket_address server_addr) {
     for (;;) {
         seastar::gate::holder holder(_gate);
+        bool shed = false;
         try {
+            semaphore_units<named_semaphore_exception_factory> units;
+            if (_conns_cpu_concurrency > 0) {
+                auto u = try_get_units(*_conns_cpu_concurrency_semaphore, 1);
+                if (u) {
+                    units = std::move(*u);
+                } else {
+                    _blocked_connections++;
+                    try {
+                    units = co_await get_units(*_conns_cpu_concurrency_semaphore, 1, std::chrono::minutes(1));
+                    } catch (const semaphore_timed_out&) {
+                        shed = true;
```

Comment:
Why do anything here? The client will close the connection if it times out, and keep it alive if not.

(will it really? how does the client measure the connection timeout?)

Reply:
we probably don't need to do `conn->shutdown` but it's easy way to free resources on our side.
`shed = true` is also for metrics, I think it'd be useful to know how many connections are timing out due to this semaphore

Reply:
Ok